avocado/utils/genio.py

```diff
@@ -174,3 +181,21 @@ def write_one_line(filename, line):
     :type line: str
     """
     write_file(filename, line.rstrip('\n') + '\n')
+
+
+def write_file_or_fail(filename, data):
+    """
+    Write to a file and raise exception on write failure
+
+    :param filename : Path to file
```

Comment:
There is not suppose to be the extra space between the object and the description:

```python
:param filename: Path to file
:type filename: str
:param data: Data to be written to a file
...
```